src/Ryujinx.HLE/HOS/Services/Ldn/UserServiceCreator/IUserLocalCommunicationService.cs

```diff
@@ -1,88 +1,1099 @@
+
+            context.ResponseData.Write((short)_disconnectReason);
+
+            return ResultCode.Success;
+        }
+
+        public void SetDisconnectReason(DisconnectReason reason)
+        {
+            if (_state != NetworkState.Initialized)
+            {
+                _disconnectReason = reason;
+
+                SetState(NetworkState.Initialized);
+            }
+        }
+
+        [CommandCmif(4)]
+        // GetSecurityParameter() -> bytes<0x20, 1> security_parameter
+        public ResultCode GetSecurityParameter(ServiceCtx context)
+        {
+            if (_nifmResultCode != ResultCode.Success)
+            {
+                return _nifmResultCode;
+            }
+
+            ResultCode resultCode = GetNetworkInfoImpl(out NetworkInfo networkInfo);
+            if (resultCode != ResultCode.Success)
+            {
+                return resultCode;
+            }
+
+            SecurityParameter securityParameter = new()
+            {
+                Data = new Array16<byte>(),
+                SessionId = networkInfo.NetworkId.SessionId
+            };
+
+            context.ResponseData.WriteStruct(securityParameter);
+
+            return ResultCode.Success;
+        }
+
+        [CommandCmif(5)]
+        // GetNetworkConfig() -> bytes<0x20, 8> network_config
+        public ResultCode GetNetworkConfig(ServiceCtx context)
+        {
+            if (_nifmResultCode != ResultCode.Success)
+            {
+                return _nifmResultCode;
+            }
+
+            ResultCode resultCode = GetNetworkInfoImpl(out NetworkInfo networkInfo);
+            if (resultCode != ResultCode.Success)
+            {
+                return resultCode;
+            }
+
+            NetworkConfig networkConfig = new()
+            {
+                IntentId = networkInfo.NetworkId.IntentId,
+                Channel = networkInfo.Common.Channel,
+                NodeCountMax = networkInfo.Ldn.NodeCountMax,
+                LocalCommunicationVersion = networkInfo.Ldn.Nodes[0].LocalCommunicationVersion,
+                Reserved2 = new Array10<byte>()
+            };
+
+            context.ResponseData.WriteStruct(networkConfig);
+
+            return ResultCode.Success;
         }
 
         [CommandCmif(100)]
         // AttachStateChangeEvent() -> handle<copy>
         public ResultCode AttachStateChangeEvent(ServiceCtx context)
         {
-            if (_stateChangeEventHandle == 0)
+            if (context.Process.HandleTable.GenerateHandle(_stateChangeEvent.ReadableEvent, out int stateChangeEventHandle) != Result.Success)
+            {
+                throw new InvalidOperationException("Out of handles!");
+            }
+
+            context.Response.HandleDesc = IpcHandleDesc.MakeCopy(stateChangeEventHandle);
+
+            // Return ResultCode.InvalidArgument if handle is null, doesn't occur in our case since we already throw an Exception.
```

Comment:
```suggestion
            // Returns ResultCode.InvalidArgument if handle is null, doesn't occur in our case since we already throw an Exception.
```
And what is "handle"? There is no variable called `handle` here.